pkgs/by-name/go/gopass-jsonapi/package.nix

```diff
@@ -0,0 +1,63 @@
+{
+  lib,
+  stdenv,
+  makeWrapper,
+  buildGoModule,
+  fetchFromGitHub,
+  installShellFiles,
+  gopass,
+  apple-sdk_15,
+  darwinMinVersionHook,
+}:
+
+buildGoModule rec {
+  pname = "gopass-jsonapi";
+  version = "1.15.15";
+
+  src = fetchFromGitHub {
+    owner = "gopasspw";
+    repo = "gopass-jsonapi";
+    rev = "v${version}";
+    hash = "sha256-nayg7NTJH6bAPiguyuN37JivfWkpOUX/xI/+PHDi3UI=";
+  };
+
+  vendorHash = "sha256-khX1CdzN+5T8q2hA3NyCxtz7uw9uDd9u61q3UslTtqs=";
+
+  subPackages = [ "." ];
+
+  nativeBuildInputs = [
+    installShellFiles
+    makeWrapper
+  ];
+
+  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
+    # For ScreenCaptureKit.h
+    apple-sdk_15
```

Comment:
How to choose the sdk version?

Reply:
@emilazy Initially I didn't try to add anything Darwin specific, but I got build errors saying:

```
# github.com/kbinani/screenshot
vendor/github.com/kbinani/screenshot/darwin.go:9:10: fatal error: 'ScreenCaptureKit/ScreenCaptureKit.h' file not found
#include <ScreenCaptureKit/ScreenCaptureKit.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

How do you explain this?

Reply:
I guess the issue is that it conditions it on `__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ > MAC_OS_VERSION_14_4`, and maybe the undefined `MAC_OS_VERSION_14_4` on our default SDK is defaulting to 0 or something? Anyway you can give it `apple-sdk_14` and omit any `darwinMinVersionHook`.

Reply:
OK thanks it worked. I wonder then, what does it mean to use `darwinMinVersionHook` along with a specific `apple-sdk` version? Why do we need the former if we are capable of choosing any of the later?